### ISSUE TITLE
EitherLogOrThrowCheck does not recognise lombock log annotations

### ISSUE THREAD

**Issue by jansauer** [sevntu-checkstyle/sevntu.checkstyle#784]
/var/tmp $ cat YOUR_FILE.java

```
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class Test {
      
    public void demo() {
        try {
          throw new RuntimeException("test");
        } catch (RuntimeException exception) {
            log.error("test", exception);
            throw new RuntimeException("test", exception);
        }
    }
}
```

/var/tmp $ cat config.xml

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="EitherLogOrThrowCheck"/>
    </module>
</module>
```

---------------

The rule violation in Test.java is not reported. 
I expect that checkstyle doesn't recognise the logger as he gets declared as class field at compile time by [lombok](https://projectlombok.org/features/log).

Interestingly adding the static final log field by hand to any class of the projects helps checkstle enough too then find `EitherLogOrThrowCheck` violations in the project.

---------------

PS: I really tried building a working cli example but was unable to do so.

**Comment by rnveach** [comment]
Since lombok modifies the bytecode in the JVM, we have so far been against making special overrides just for it. We deal with the source code as it is.

See https://github.com/checkstyle/checkstyle/issues/5470 .

@romani ping

**Comment by jansauer** [comment]
@rnveach that confirms my assumption.

How about, instead of extending the check to recognise lombok / one liberty the `EitherLogOrThrowCheck` could have a parameter to tread all class fields with name X as loggers.

Also thx for the issue reference.

**Comment by romani** [comment]
> PS: I really tried building a working cli example but was unable to do so.

please share problems that run into, we do want to ease issue creation process, we appreciate feedback.
Did you fail to download jars ?

-----

checkstyle is sourcecode analyzer (not a bitecode), so it works badly with bycode generation tools. Lombok is inly one of them, if we allow lombok there will another issue to allow something else.
Lombok can probably change/extend behavior of annotation and we will have to deal with different versions and .... .

As this is not a main checkstyle project, we have more freedom to experiments. 
What if we will just create new Check based on EitherLogOrThrowCheck and name it `EitherLogOrThrowLombokCheck` and let it validate code as you suggested. So you will use new Check and all other users will not be affected.
